### ISSUE TITLE
Content-Security-Policies

### ISSUE THREAD

**Issue by justrealmilk** [directus/directus#12493]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues?q=is%3Aissue) that already describes my problem.

### Describe the Bug

After #12420, I dug into my NGINX config and tried a lot of things until I realised I was querying the Directus root (localhost) and comparing it to the response from an asset (/assets/) path. As it turns out, I was never the problem 🌟

yml config keys have no effect on Content-Security-Policy headers for assets

Code points:
https://github.com/directus/directus/blob/main/api/src/app.ts#L95
https://github.com/directus/directus/blob/main/api/src/controllers/assets.ts#L112

### To Reproduce

Set up a Docker container with
```yml
      CONTENT_SECURITY_POLICY_DIRECTIVES__MEDIA_SRC: "array:'self',https://directus.upliftnaturereserve.com,https://patreon.com"
      # ASSETS_CONTENT_SECURITY_POLICY: "'self' https://directus.upliftnaturereserve.com https://patreon.com"
      ASSETS_CONTENT_SECURITY_POLICY: '{"directives":{"defaultSrc":["self","https://directus.upliftnaturereserve.com","https://patreon.com"]}}'
```

### Errors Shown

curl http://localhost:8057 -v
```
*   Trying 127.0.0.1:8057...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8057 (#0)
> GET / HTTP/1.1
> Host: localhost:8057
> User-Agent: curl/7.68.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 302 Found
< Content-Security-Policy: script-src 'self' 'unsafe-eval';worker-src 'self' blob:;child-src 'self' blob:;img-src 'self' data: blob: https://cdn.directus.io;media-src 'self' https://directus.upliftnaturereserve.com https://patreon.com;connect-src 'self' https://*;default-src 'self';base-uri 'self';block-all-mixed-content;font-src 'self' https: data:;frame-ancestors 'self';object-src 'none';script-src-attr 'none';style-src 'self' https: 'unsafe-inline'
< X-Powered-By: Directus
< Vary: Origin, Accept
< Access-Control-Allow-Credentials: true
< Access-Control-Expose-Headers: Content-Range
< Location: ./admin
< Content-Type: text/plain; charset=utf-8
< Content-Length: 29
< Date: Wed, 30 Mar 2022 22:35:28 GMT
< Connection: keep-alive
< Keep-Alive: timeout=5
<
* Connection #0 to host localhost left intact
* 
```

curl http://localhost:8057/assets/d96dc263-2991-4788-82d5-00d68162c634f -v
```
*   Trying 127.0.0.1:8057...
* TCP_NODELAY set
* Connected to localhost (127.0.0.1) port 8057 (#0)
> GET /assets/d96dc263-2991-4788-82d5-00d68162c634f HTTP/1.1
> Host: localhost:8057
> User-Agent: curl/7.68.0
> Accept: */*
>
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Content-Security-Policy: default-src none
< X-Powered-By: Directus
< Vary: Origin
< Access-Control-Allow-Credentials: true
< Access-Control-Expose-Headers: Content-Range
< Content-Type: video/mp4
< Accept-Ranges: bytes
< Cache-Control: public, max-age=2592000
< Content-Length: 48063348
< Date: Wed, 30 Mar 2022 22:36:12 GMT
< Connection: keep-alive
< Keep-Alive: timeout=5
<
Warning: Binary output can mess up your terminal. Use "--output -" to tell
Warning: curl to output it to your terminal anyway, or consider "--output
Warning: <FILE>" to save to a file.
* Failed writing body (0 != 14097)
* Closing connection 0
```

### What version of Directus are you using?

9.7.1

### What version of Node.js are you using?

14.17.3

### What database are you using?

--

### What browser are you using?

--

### What operating system are you using?

--

### How are you deploying Directus?

Docker

**Comment by azrikahar** [comment]
Thanks for clarifying it's the ASSETS CSP in your scenario 😄

The following seems to work for me, would you mind trying it on your end?

```yaml
ASSETS_CONTENT_SECURITY_POLICY_DIRECTIVES__MEDIA_SRC="array:'self',https://directus.upliftnaturereserve.com,https://patreon.com"
```

![chrome_o0ugIzAXZn](https://user-images.githubusercontent.com/42867097/161006144-73619fe6-3d4e-4480-81c2-d1e0040e8c55.png)

---

Unfortunately I think the JSON approach is bugged, so try not to use that syntax for now. Once we verified `ASSETS_CONTENT_SECURITY_POLICY` itself is working when you use the `array:` syntax, we'll use another ticket to track the env JSON parsing issue.

**Comment by justrealmilk** [comment]
That's it! I never tried that because the pattern I saw in the code didn't match my expectations so I thought there was only a single assets key. Better add this to the docs!

**Comment by benhaynes** [comment]
CC @erondpowell 🙂
